source/nodejs/adaptivecards/src/card-elements.ts

```diff
@@ -6537,6 +6553,16 @@ export class ActionSet extends CardElement {
         return super.findDOMNodeOwner(node);
     }
 
+    getElementById(id: string): CardElement | undefined {
+        let result = super.getElementById(id);
+
+        if (!result) {
+            result = this.getElementByIdFromAction(id);
+        }
+
+        return result;
```

Comment:
this looks better:)
do we absolutely have to go over the tree **twice** in this scenario?
can we somehow detect this scenario in another way?

Reply:
Do you mean call `super.getElementById` and `this.getElementByIdFromAction`?

Reply:
If so, `getElementById` doesn't normally traverse the tree. It checks if the current element is the correct id :)

Reply:
oh I see... the name of the method - `getElementById` doesn't reflect what it does then :D